src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -363,7 +363,7 @@ public override bool AreLocalsZeroed
             get
             {
                 var data = this.GetDecodedWellKnownAttributeData();
-                return data != null ? !data.HasSkipLocalsInitAttribute : true;
+                return data != null ? !data.HasSkipLocalsInitAttribute : ContainingType.AreLocalsZeroed;
```

Comment:
>return data != null ? !data.HasSkipLocalsInitAttribute : ContainingType.AreLocalsZeroed; [](start = 16, length = 88)

The logic here feels inaccurate. I think that, even if we have ```data```, we should still delegate to when ```data.HasSkipLocalsInitAttribute``` is false.

Reply:
I didn't get what you meant. When data is not null, we use it. When it is null, we delegate it to the containing type.

Reply:
> I didn't get what you meant. When data is not null, we use it. When it is null, we delegate it to the containing type.

The presence of ```data``` doesn't mean there was a SkipLocalsInitAttribute on the method. In that case we should still delegate to the type. Please add a test when there is some other well known attribute on the method

---
In reply to: [175588950](https://github.com/dotnet/roslyn/pull/25276#discussion_r175588950) [](ancestors = 175588950)

Reply:
Now I get it. You're right.

Reply:
I added a test, but we can only truly test the scenario we want once we enable applying SkipLocalsInitAttribute on classes. That would be when the type contains the attribute and the method does not. I put a prototype comment as a reminder.